### PR TITLE
docs:  resource groups sql ref pages - create new, updates to others

### PR DESCRIPTION
**Pull request by lisakowen**
first pass at new and updated SQL reference pages for the resource groups feature.

to see nicely formatted versions, here are links to the new/updated SQL ref pages on a review staging site.  (note: this site is a work-in-progress staging area, so please do not use as a go-to source for gpdb 5.0 docs.)
- http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/ALTER_ROLE.html
- http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/CREATE_ROLE.html
- http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/ALTER_RESOURCE_GROUP.html (new)
- http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html (new)
- http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/DROP_RESOURCE_GROUP.html (new)

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      <title>Parameters</title>
+      <parml>
+        <plentry>
+          <pt>
+            <varname>name</varname>
+          </pt>
+          <pd>The name of the resource group.</pd>
+        </plentry>
+        <plentry>
+          <pt>CONCURRENCY <varname>integer</varname></pt>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: Repeat previous edits for CONCURRENCY and CPU_RATE_LIMIT.  Also, if I'm reading this correctly those are the only two values that can be ALTERed later.  If so it probably makes sense to mention that here.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          <pt>CPU_RATE_LIMIT <varname>integer</varname></pt>
+          <pd>Required. Sets the total percentage of Greenplum Database CPU resources to allocate to this resource group. The minimum CPU percentage you can specify for a resource group is 1. The maximum is 100. The sum of the <codeph>CPU_RATE_LIMIT</codeph> values specified for all resource groups defined in the Greenplum Database cluster must be less than or equal to 100.
+          </pd>
+        </plentry>
+        <plentry>
+          <pt>MEMORY_LIMIT <varname>integer</varname></pt>
+          <pd>Required. Sets the total percentage of Greenplum Database memory resources to allocate to all transactions submitted from roles assigned this resource group. The minimum memory percentage you can specify for a resource group is 1. The maximum is 100. The sum of the <codeph>MEMORY_LIMIT</codeph> values specified for all resource groups defined in the Greenplum Database cluster must be less than or equal to 100.</pd>
+        </plentry>
+        <plentry>
+          <pt>MEMORY_SHARED_QUOTA <varname>integer</varname></pt>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: I don't fully understand how MEMORY_SHARED_QUOTA and MEMORY_SPILL_RATIO work from the descriptions, and there are no examples included below.  Let's discuss this more and consider some edits.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        </plentry>
+        <plentry>
+          <pt>MEMORY_SPILL_RATIO <varname>integer</varname></pt>
+          <pd>The <codeph>MEMORY_SPILL_RATIO</codeph> identifies how much more or less memory to allocate to a transaction before spilling to disk. The minimum memory spill ratio percentage you can specify for a resource group is 1. The maximum is 100. The upper limit is (100 - <codeph>MEMORY_SHARED_QUOTA</codeph>). The default <codeph>MEMORY_SPILL_RATIO</codeph> value is 20. The sum of the <codeph>MEMORY_SHARED_QUOTA</codeph> and <codeph>MEMORY_SPILL_RATIO</codeph> values specified for a resource group cannot exceed 100.</pd>
+        </plentry>
+      </parml>
+    </section>
+    <section id="section5"><title>Notes</title>
+        <p>You cannot submit an <codeph>CREATE RESOURCE GROUP</codeph> command in a sub-transaction.</p>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: "an" -> "a"

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        <p>Use the
+          <codeph>gp_toolkit.gp_resgroup_config</codeph> system view to display the limit
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: Make this view ame into an XREF if possible.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_ROLE.xml [hunk 5]
```diff
@@ -156,6 +157,23 @@
                         giving them the right to grant membership in this role to others.</pd>
                 </plentry>
                 <plentry>
+                    <pt>RESOURCE GROUP <varname>group_name</varname></pt>
+                    <pd>When resource group-based workload management is enabled,
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_ROLE.xml]: Repeat edits from ALTER ROLE.

File: gpdb-doc/dita/ref_guide/sql_commands/DROP_RESOURCE_GROUP.xml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        to a specific user/role.</p><p> Perform the following query to view all of the currently active
+        queries for all resource groups:</p><codeblock>SELECT usename, current_query, waiting, procpid,
+    rsgid, rsgname, rsgqueueduration 
+  FROM pg_stat_activity;
+</codeblock><p>To view the resource group assignments, perform the following query
+on the <codeph>pg_roles</codeph> and <codeph>pg_resgroup</codeph> system
+catalog tables:</p><codeblock>SELECT rolname, rsgname 
+  FROM pg_roles, pg_resgroup
+  WHERE pg_roles.rolresgroup=pg_resgroup.oid;</codeblock></section><section id="section6"><title>Examples</title>
+<p>Remove the resource group named <codeph>adhoc</codeph>:</p><codeblock>DROP RESOURCE GROUP adhoc;</codeblock></section>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/DROP_RESOURCE_GROUP.xml]: Seems like the order of these example commands should be reversed.

File: gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml [hunk 7]
```diff
@@ -44,7 +46,13 @@ its password if the password is MD5-encrypted.</li><li id="av136604"><b>SET | RE
           role can only belong to one resource queue. For a role without <codeph>LOGIN</codeph>
           privilege, resource queues have no effect. See <codeph><xref
               href="./CREATE_RESOURCE_QUEUE.xml#topic1" type="topic" format="dita"/></codeph> for
-          more information.</li><li id="av137061"><b>WITH <varname>option</varname></b> — Changes many of the role attributes that can be
+          more information.</li><li id="av1370583"><b>RESOURCE GROUP</b> — When resource group-based workload management is configured, assigns a resource group to the role.
+          The role would then be subject to the concurrent transaction, memory, and CPU limits
+          configured for the resource group. You can
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml]: that sounds a lot better!

File: gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml [hunk 8]
```diff
@@ -44,7 +46,13 @@ its password if the password is MD5-encrypted.</li><li id="av136604"><b>SET | RE
...
+          assign a single resource group to a role, and you can assign a resource group to multiple
+          roles. See <codeph><xref
+              href="./CREATE_RESOURCE_GROUP.xml#topic1" type="topic" format="dita"/></codeph> for
+          additional information.</li><li id="av137061"><b>WITH <varname>option</varname></b> — Changes many of the role attributes that can be
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml]: good question!  i just tried it out (resource groups not enabled) and you can set both.  but you get a warning that resource group is disabled...

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml (hunk 1 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: well, you will be able to alter some of the memory* limits, but that hasn't made it to the code yet.  was keeping the pages to what is currently implemented.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml (hunk 2 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: you and me both.  hoping the engineering team can provide more info here.  

the Memory section of the Resource Group wiki page is currently empty.  wanted to get some base info in these pages and build on it later.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml (hunk 4 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: will do this in a later PR.  (haven't submitted the PR for the view yet.)

**Comment by lisakowen** [comment]
thank you for reviewing, @dyozie .  will touch base after engineering review of this content.

File: gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml [hunk 9]
```diff
@@ -44,7 +46,12 @@ its password if the password is MD5-encrypted.</li><li id="av136604"><b>SET | RE
           role can only belong to one resource queue. For a role without <codeph>LOGIN</codeph>
           privilege, resource queues have no effect. See <codeph><xref
               href="./CREATE_RESOURCE_QUEUE.xml#topic1" type="topic" format="dita"/></codeph> for
-          more information.</li><li id="av137061"><b>WITH <varname>option</varname></b> — Changes many of the role attributes that can be
+          more information.</li><li id="av1370583"><b>RESOURCE GROUP</b> — Assigns a resource group to the role.
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml]: Any reason the first sentence for Resource Queue/Group are reversed
--role to queue
--group to role

**Comment by mkiyama** [comment]
Roles can be assigned to both a queue and group?
If so, might want to mention that and that only one is active at a time.

**Comment by mkiyama** [comment]
In create resource group
Are there defaults for  
MEMORY_LIMIT
CPU_RATE_LIMIT

Also, you might want to mention that only resource groups or queues are active, not both,
and how to change.

CPU_RATE_LIMIT - Isn't some external OS config required for CPU (cgroups)?
Might want to mention that.

Might want to mention something in create resource group if that isn't in another doc task.

File: gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml (hunk 9 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/ALTER_ROLE.xml]: i did not modify any of the resource queue content in this PR.  most of the RQ content is written "assign role to queue".  i am using what i think is correct for RGs - you assign an RG to a role, not the other way around.

**Comment by lisakowen** [comment]
@mkiyama - re: assigning both an RQ and RG to a role - looks like the code currently allows this.  will think about how to handle this throughout the docs, not just here.  i removed "when RG-based workload management is enabled" from some of the content.  this might flush itself out when all of the content is in place ...

**Comment by lisakowen** [comment]
@mkiyama - re: cgroups config for CPU resource mgmt.  yes, external config is required here.  i am currently writing the content for this section.  will add a link or some other qualifier about this.

thanks for reviewing.

**Comment by lisakowen** [comment]
queries vs. transactions vs. statements - what terminology is accurate?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+         group is assigned to one or more roles and identifies concurrent transaction, memory,
+         and CPU limits for the role when resource group-based workload management is enabled. Only a superuser can create a resource group.</p>
+      <p>Greenplum Database pre-defines two default resource groups: <codeph>admin_group</codeph>
+         and <codeph>default_group</codeph>. These group names are reserved, as is the group name <codeph>NONE</codeph>.</p>
+      <p>The <codeph>max_resource_groups</codeph> server configuration parameter governs the maximum number of resource groups you can create.</p>
+      <p>
+        To set appropriate limits for resource groups, the Greenplum Database administrator must
+        be familiar with the queries typically executed on the system, as well as the users/roles executing those queries.</p>
+      <p>
+        After defining a resource group, assign the group to one more more roles using the <codeph><xref
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: typo. should be one  or more?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+          <pd>Required. The total percentage of Greenplum Database memory resources to allocate to this resource group. The minimum memory percentage you can specify for a resource group is 1. The maximum is 100. The sum of the <codeph>MEMORY_LIMIT</codeph> values specified for all resource groups defined in the Greenplum Database cluster must be less than or equal to 100.</pd>
+        </plentry>
+        <plentry>
+          <pt>MEMORY_SHARED_QUOTA <varname>integer</varname></pt>
+          <pd>The quota of shared memory in the resource group. Resource groups with a <codeph>MEMORY_SHARED_QUOTA</codeph> threshold set aside a percentage of memory allotted to the resource group to share across transactions. This shared memory isallocated on a first-come first-server basis as available. A transaction may use none, some, or all of this memory. The minimum memory shared quota percentage you can specify for a resource group is 0. The maximum is 100. The default <codeph>MEMORY_SHARED_QUOTA</codeph> value is 20. The sum of the <codeph>MEMORY_SHARED_QUOTA</codeph> and <codeph>MEMORY_SPILL_RATIO</codeph> values specified for a resource group cannot exceed 100.</pd>
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: typo. isallocated should be is allocated?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        </plentry>
+        <plentry>
+          <pt>MEMORY_SPILL_RATIO <varname>integer</varname></pt>
+          <pd>The memory usage threshold for memory-intensive operators in a transaction. When the transaction reaches this threshold, it spills to disk. The minimum memory spill ratio percentage you can specify for a resource group is 1. The maximum is 100. The upper limit is (100 - <codeph>MEMORY_SHARED_QUOTA</codeph>). The default <codeph>MEMORY_SPILL_RATIO</codeph> value is 20. The sum of the <codeph>MEMORY_SHARED_QUOTA</codeph> and <codeph>MEMORY_SPILL_RATIO</codeph> values specified for a resource group cannot exceed 100.</pd>
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: memory_spill_ratio default value is 30

File: gpdb-doc/dita/ref_guide/sql_commands/DROP_RESOURCE_GROUP.xml [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1"><title id="dc20941">DROP RESOURCE GROUP</title><body><p id="sql_command_desc">Removes a resource group.</p><section id="section2"><title>Synopsis</title><codeblock id="sql_command_synopsis">DROP RESOURCE GROUP <varname>group_name</varname></codeblock></section><section id="section3"><title>Description</title><p>This command removes a resource group from Greenplum
+Database. Only a superuser
+can drop a resource group.</p><p>To drop a resource group, the group cannot be assigned to any roles,
+nor can it have any statements pending in the group.</p><p>You cannot drop the pre-defined <codeph>admin_group</codeph> and <codeph>default_group</codeph> resource groups.</p>
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/sql_commands/DROP_RESOURCE_GROUP.xml]: pending or running in the group

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml (hunk 12 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: i found the 20 value in master branch, backend/commands/resgroupcmds.c:
    #define RESGROUP_DEFAULT_MEM_SPILL_RATIO (20)

is this default being updated elsewhere in the code or in a pending or future PR that i missed?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_ROLE.xml [hunk 14]
```diff
@@ -156,6 +157,21 @@
...
+                    <pt>RESOURCE GROUP <varname>group_name</varname></pt>
+                    <pd> The name of the resource group to assign to the the new role.  The role 
+                        will be subject to the concurrent transaction, memory, and CPU limits
+                        configured for the resource group.
+                        You can assign a single resource group to one or more roles.</pd>
+                    <pd>If you do not specify a resource group for a new role, the role is
+                        automatically assigned the default resource group for the role's capability,
+                        <codeph>admin_group</codeph> for <codeph>SUPERUSER</codeph> roles,
+                        <codeph>default_group</codeph> for non-admin roles.</pd>
+                    <pd>You may assign a role with the <codeph>SUPERUSER</codeph> attribute the
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_ROLE.xml]: For the last two paragraphs, would something like this be clearer? 
You can specify any role to either default resource group, admin_group or default_group.

Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_ROLE.xml]: i think addressing the default groups in separate statements is clearer.  i updated the text to better clarify.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+[ MEMORY_SPILL_RATIO=<varname>integer</varname> ]</codeblock>
+    </section>
+    <section id="section3">
+      <title>Description</title>
+      <p>Creates a new resource group for Greenplum Database workload management. A resource
+         group is assigned to one or more roles and identifies concurrent transaction, memory,
+         and CPU limits for the role when resource group-based workload management is enabled. Only a superuser can create a resource group.</p>
+      <p>Greenplum Database pre-defines two default resource groups: <codeph>admin_group</codeph>
+         and <codeph>default_group</codeph>. These group names are reserved, as is the group name <codeph>NONE</codeph>.</p>
+      <p>The <codeph>max_resource_groups</codeph> server configuration parameter governs the maximum number of resource groups you can create.</p>
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: Might want to have short description of what NONE is used for.

Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.xml]: i removed the reference to NONE.  it is applicable only in reference to resource group reassignment (ALTER ROLE).

**Pull request merged**